grails-app/utils/au/org/emii/portal/HtmlCharacterEntityReferences.groovy

```diff
@@ -19,7 +29,7 @@ final class HtmlCharacterEntityReferences {
      * @since 4.1.2
      */
     public String convertToReference(char character, String encoding) {
-        if (encoding.startsWith("UTF")){
+        if (encoding.startsWith("UTF-")){
```

Comment:
The return type from getEncoding() from an input stream was UTF8, while the value contained in the header is "UTF-8" (which is a better match for the internal groovy definition of the charset). I've taken the opportunity to tighten this conditional